uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -386,6 +424,10 @@ define([
        * @returns {Promise}
        */
       function initDayTypesAndStatus() {
+        if (vm.leaveRequestDayTypes.length && vm.leaveRequestStatuses.length) {
```

Comment:
future optimization may need to call the function again and again from say `onDateChange` and then this check will be useful